file/writer.go

```diff
@@ -605,6 +633,36 @@ func populateConsumers(kongState *state.KongState, file *Content,
 	return nil
 }
 
+func setPluginSharedConfig(fp *FPlugin, sharedPlugins map[string]utils.SharedPlugin,
+	consumerID, serviceID, routeID string) {
+	for name, content := range sharedPlugins {
+		for _, consumer := range content.Consumers {
+			if consumer != consumerID {
+				continue
+			}
+			fp.Plugin.Config = nil
+			fp.ConfigSource = kong.String(name)
+			break
+		}
+		for _, service := range content.Services {
+			if service != serviceID {
+				continue
+			}
+			fp.Plugin.Config = nil
+			fp.ConfigSource = kong.String(name)
+			break
+		}
+		for _, route := range content.Routes {
+			if route != routeID {
+				continue
+			}
+			fp.Plugin.Config = nil
+			fp.ConfigSource = kong.String(name)
+			break
+		}
+	}
```

Comment:
Suggest two functions: one takes in the plugin map and the relation IDs, and returns the name of the plugin config source (if found), and this one then sets the config to nil and source to non-nil. This way you can unit-test the function that looks through sharedPlugins map.

Reply:
refactored a bit